### DISCUSSION TITLE
Encryption of SMTP Password

### DISCUSSION THREAD

**Discussion by SharperThanThou** [dani-garcia/vaultwarden#3619] [General]
Hello!  Very recently I updated the admin page token to leverage the hashed token via vaultwarden hash.  Is the same possible for the SMTP_PASSWORD configuration value?  I would prefer to not store any configuration values in plaintext.

Note that I attempted to create the vaultwarden hash value of my SendGrid API key and use that argon2id PHC string as the SMTP_PASSWORD without success (test emails threw a 535):

![image](https://github.com/dani-garcia/vaultwarden/assets/40498174/8c4ef219-0f35-448e-9ef2-6710be957657)

Much appreciated!

**Comment by BlackDex** [comment]
We need the plain text representative of the password, because that needs to be sent to the server. Also using a hashed value which would work during login doesn't make it safe, since anybody else then can use that too.
